packages/react-utilities/src/utils/applyTriggerPropsToChildren.ts

```diff
@@ -11,8 +11,25 @@ export const applyTriggerPropsToChildren = <TTriggerProps>(
   if (typeof children === 'function') {
     return children(triggerProps);
   } else if (children) {
-    return React.cloneElement(onlyChild(children), triggerProps);
+    return cloneTriggerTree(children, triggerProps);
   }
 
   return children;
 };
+
+/**
+ * Clones a React element tree, and applies the given props to the first grandchild that is not
+ * a FluentTriggerComponent or React Fragment (the same element returned by {@link getTriggerChild}).
+ */
+const cloneTriggerTree = <TTriggerProps>(child: React.ReactNode, triggerProps: TTriggerProps): React.ReactElement => {
+  if (!React.isValidElement(child)) {
+    throw new Error(`Component's child must be a single element`);
+  }
+
+  if (child.type === React.Fragment || isFluentTriggerComponent(child.type)) {
```

Comment:
```suggestion
  if (isFluentTriggerComponent(child.type)) {
```

I don't see any reason to support React fragments, as we will throw in this scenario:

```tsx
function App() {
  return (
    <Tooltip>
      <>
        <Button />
        <Button />
      </>
    </Tooltip>
  );
}
```

Reply:
The alternative is to throw if we find a Fragment, since Fragments don't support any props. Is that preferrable?

```ts
 if (!React.isValidElement(child) || child.type === React.Fragment) {
    throw new Error(`Component's child must be a single element, and not a Fragment`);
}
```

Reply:
> Is that preferrable?

Yes, we can use `React.Children.only()` and then additionally check for `React.Fragment`.

Reply:
Ok I changed it to throw for Fragments.